doc/developer/scripting.rst

```diff
@@ -53,159 +53,297 @@ Reasons against supporting multiple scripting languages:
+Output
+""""""
 
+.. code-block:: c
 
-Querying State
-^^^^^^^^^^^^^^
+   int a = 100, b = 200, c = 300;
+   frrscript_call(fs, "on_foo", ("a", &a), ("b", &b), ("c", c));
+   // a is 500, b is 200, c is 300
+
+   int* d = frrscript_get_result(fs, "on_foo", "d", lua_tointegerp);
+   // d is 800
+
+
+.. code-block:: lua
+
+   function on_foo(a, b, c)
+     b = 600
+     return { ["a"] = 500, ["c"] = 700, ["d"] = 800 }
+   end
+
+
+**Lua functions being called must return a single table of string names to
+values.**
+(Lua functions should return an empty table if there is no output.)
+The keys of the table are mapped back to names of variables in C. Note that
+the values in the table can also be tables. Since tables are Lua's primary
+data structure, this design lets us return any Lua value.
+
+After the Lua function returns, the names of variables  to ``frrscript_call()``
+are matched against keys of the returned table, and then decoded. The types
+being decoded must have registered decoders (more below); the compiler will
+warn you otherwise.
+
+In the example, since ``a`` was in the returned table and ``b`` was not,
+``a`` was decoded and its value modified, while ``b`` was not decoded.
+``c`` was decoded as well, but its decoder is a noop.
+What modifications happen given a variable depends whether its name was
+in the returned table and the decoder's implementation.
+
+.. warning::
+   Always keep in mind that non const-qualified pointers in
+   ``frrscript_call()`` may be modified - this may be a source of bugs.
+   On the other hand, const-qualified pointers and other values cannot
+   be modified.
+
+
+.. tip::
+   You can make a copy of a data structure and pass that in instead,
+   so that modifications only happen to that copy.
+
+``frrscript_call()`` returns 0 if and only if the Lua function was successfully
+called. A non-zero return could indicate either a missing Lua script, a missing
+Lua function, or an error from the Lua interpreter.
 
-.. todo::
+In the above example, ``d`` was not an input to ``frrscript_call()``, so its
+value must be explicitly retrieved with ``frrscript_get_result``.
 
-   This section will be updated once ``frrscript_get_result`` has been
-   updated to work with the new ``frrscript_call`` and the rest of the new API.
+``frrscript_get_result()`` takes a
+decoder and string name which is used as a key to search the returned table.
+Returns the pointer to the decoded value, or NULL if it was not found.
+In the example, ``d`` is a "new" value in C space,
+so memory allocation might take place. Hence the caller is
+responsible for memory deallocation.
 
 
-Unloading
-^^^^^^^^^
+Delete
+^^^^^^
 
-To destroy a script and its associated state:
+To delete a script and the all Lua states associated with it:
 
 .. code-block:: c
 
-   frrscript_unload(fs);
+   frrscript_delete(fs);
+
+
+A complete example
+""""""""""""""""""
+
+So, a typical exection call, with error checking, looks something like this:
```

Comment:
exection?

Reply:
fixed in https://github.com/FRRouting/frr/pull/8982/commits/fe80e6019b5fbca6d3acf4d3be92f45f4a53ad18